dask/array/core.py

```diff
@@ -4165,13 +4167,13 @@ def asarray(a, allow_unknown_chunksizes=False, *, like=None, **kwargs):
         if like is not None:
             if not _numpy_120:
                 raise RuntimeError("The use of ``like`` required NumPy >= 1.20")
-            a = np.asarray(a, like=meta_from_array(like))
+            a = np.asarray(a, like=meta_from_array(like), dtype=dtype, order=order)
         else:
-            a = np.asarray(a)
+            a = np.asarray(a, dtype=dtype, order=order)
     return from_array(a, getitem=getter_inline, **kwargs)
 
 
-def asanyarray(a, *, like=None):
+def asanyarray(a, dtype=None, order=None, *, like=None):
```

Comment:
Similar comment here re: docstring

Reply:
yeah totally! I might just steal the docstring from numpy though via a `@derived_from`

Reply:
In general I'm in favor of using `derived_from`, though in this case we have an extra `allow_unknown_chunksizes` keyword and it looks like `like` has some Dask-specific verbiage. Maybe manually copying over the descriptions from NumPy makes more sense here?

Reply:
Actually I just copied them over since the `like` is already customized.